pygmt/modules.py

```diff
@@ -214,3 +217,85 @@ def __exit__(self, exc_type, exc_value, traceback):
         )
         with Session() as lib:
             lib.call_module("set", arg_str)
+
+
+@xr.register_dataarray_accessor("gmt")
+class GMTDataArrayAccessor:
+    """
+    This is the GMT extension for :class:`xarray.DataArray`.
+
+    You can access various GMT specific metadata as follow:
+
+    >>> from pygmt.datasets import load_earth_relief
+    >>> # Use the global Earth relief grid with 1 degree spacing
+    >>> grid = load_earth_relief(resolution="01d")
+
+    >>> # See if grid uses Gridline (0) or Pixel (1) registration
+    >>> grid.gmt.registration
+    0
+    >>> # See if grid uses Cartesian (0) or Geographic (1) coordinate system
+    >>> grid.gmt.gtype
+    1
+    """
+
+    def __init__(self, xarray_obj):
+        self._obj = xarray_obj
+        try:
+            self._source = self._obj.encoding["source"]
+            self._info = grdinfo(self._source)
```

Comment:
I know we're not trying to maintain GMT 6.0 compatibility anymore, and I've had a look at how to refactor the code using `grdinfo -C`. However, the code becomes very unreadable, because we're trying to index the string output from `grdinfo`. 

```suggestion
            self._info = grdinfo(self._source)
            self._registration = int(grdinfo(self._source, C="n", o=10))
            self._gtype = int(grdinfo(self._source, C="n", o=11))
```

The best I can come up with is to add the two lines here, so that `xarray.DataArray` grids **with** a NetCDF source can return the `registration` and `gtype` info quickly, but grids **without** a source take a small one-time penalty with the string parsing, e.g. `if "Gridline node registration used" in self._info: self._registration = 0`.

To be honest, it might not be any faster even, since `grdinfo` will be run 3 times. Or maybe there's another way to refactor this for GMT 6.1 that I'm missing?

Reply:
I'm not sure if I understand your codes correctly. Can we just run:
```python
try:
       self._source = self._obj.encoding["source"]
       self._registration = int(grdinfo(self._source, C="n", o=10))
       self._gtype = int(grdinfo(self._source, C="n", o=11))
except KeyError:
       self._registration = 0
       self._gtype = 0
```

Reply:
The two grdinfo calls can also be reduce into one single call, like:
```python
self._registration, self._gtype = map(int, grdinfo(self._source, C="n", o=10,11).split())
```

I didn't try to run the code above. It may not work, but the idea is correct.

Reply:
Yes this works, it passes all the complicated tests I wrote at least. Though it might not be very readable.

Reply:
Adding some comments can make it more readable. I think we may refactor the output of `grdinfo()` later.

BTW, `o=10,11` should be `o="10,11"`

Reply:
Cool, will make a commit before I head out later. Agree that we can refactor later to add in more useful info, I think Liam wanted to improve it at https://github.com/GenericMappingTools/pygmt/issues/147#issuecomment-450018956.

Reply:
Alright done. I'll be away for the rest of the day. Feel free to merge this one in (or make changes as you see fit) **once** #507 is merged into master. I'm not sure how we're going to handle the Scipy Sprint with so many tests breaking on GMT 6.1 but I suppose we'll have to make do.